### PR TITLE
refactored api.js to axios and removed jquery (#186)

### PR DESCRIPTION
**Pull request by marc-maurer**
This pullrequest removes jquery from API.js and replaced it with axios.
Issue #186

### CODE REVIEW

**Comment by Machine-Maker** [comment]
This is just a syntax-type suggestion, but I know I always like to, in axios promises, instead of doing 
```javascript
axios.get().then((response) => {
    let data = response.data;
    //...
});
```
I like
```javascript
axios.get().then(({ data }) => {
    //...
});
```
idk if that's a bad practice or smth, feel free to correct me.

**Comment by marc-maurer** [comment]
Done! Also found and fixed a bug that I accidentally introduced with my first change.

File: src/main/frontend/src/api.js [hunk 1]
```diff
@@ -50,45 +40,38 @@ export class API {
...
                 if (session === null || (!isNaN(new Date(session.expires).getTime()) && new Date(session.expires) < date)) {
-                    return $.ajax({
-                        url: '/api/v1/authenticate/user',
-                        method: 'POST',
-                        dataType: 'json',
-                        contentType: 'application/json',
-                    })
-                        .done((data) => {
+                    return axios
+                        .post('/api/v1/authenticate', {}, { headers: { 'Content-Type': 'application/json' } })
```
Comment (MiniDigger) [inline src/main/frontend/src/api.js:44]: Why does this not post against authenticate/user anymore?

File: src/main/frontend/src/api.js [hunk 2]
```diff
@@ -50,45 +40,38 @@ export class API {
...
-                    })
-                        .done((data) => {
+                    axios
+                        .post('/api/v1/authenticate', {}, { headers: { 'Content-Type': 'application/json' } })
+                        .then(({ data }) => {
                             if (data.type !== 'public') {
                                 reject('Expected public session from public authentication');
                             } else {
                                 localStorage.setItem('public_api_session', JSON.stringify(data));
+                                console.log(data);
```
Comment (MiniDigger) [inline src/main/frontend/src/api.js:69]: Muh debug logging spam

File: src/main/frontend/src/components/DependencySelection.vue [hunk 3]
```diff
@@ -297,8 +297,8 @@ export default {
             for (const dep of deps) {
                 if (dep.project_id) {
                     this.dependencyLinking[platformName][dep.name] = 'Hangar';
-                    API.request(`projects/${dep.project_id}`).then((res) => {
-                        this.selectProject(platformName, dep.name, res);
+                    API.request(`projects/${dep.project_id}`).then((response) => {
+                        this.selectProject(platformName, dep.name, response.data);
                     });
```
Comment (MiniDigger) [inline src/main/frontend/src/components/DependencySelection.vue:302]: Could prolly use a destructing pattern here too

**Review by MiniDigger — CHANGES_REQUESTED** [review]
Two minor things, else looks nice, thanks!

**Pull request merged**